homeassistant/components/tplink/switch.py

```diff
@@ -98,6 +96,7 @@ def turn_on(self, **kwargs):
     def turn_off(self, **kwargs):
         """Turn the switch off."""
         self.smartplug.turn_off()
+        self.update_state()
```

Comment:
This should not be needed? Core will call `update()` to update the state after the service has completed.

Reply:
It should not be needed, I am removing it.